Main/Shared/FileSystem/FileSystem.h

```diff
@@ -0,0 +1,112 @@
+// Created by Mark "hogsy" Sowden, 2023-2025 <hogsy@snortysoft.net>
+// https://oldtimes-software.com/jaded/
+
+#pragma once
+
+#if defined( __cplusplus )
+
+namespace jaded
+{
+	class FileSystem
+	{
+	public:
+		typedef uint32_t Key;
+
+		typedef uint32_t FileIndex;
+		typedef uint32_t DirIndex;
+
+		struct KeyFile
+		{
+			std::string name;   // name of the file, without directory (see dir)
+			DirIndex    dir;    // index into the directory table
+			FileIndex   index;  // index into the files table
+			FileIndex   bfIndex;// used during extraction
+			Key         key;    // key
+
+			bool markedForDeletion;
+		};
+
+		struct KeyDir
+		{
+			std::string name;
+			DirIndex    index;
+
+			std::vector< FileIndex > files;
+		};
+
+		FileSystem()  = default;
+		~FileSystem() = default;
+
+		std::string GetExecutablePath();
+		std::string GetAppDataPath();
+		std::string NormalizePath( std::string path );
```

Comment:
Some of these methods feel like they might be better being static rather than needing an instance.